### ISSUE TITLE
When playing a directory of music files, IINA regularly skips to a (random?) time when moving to the next track.

### ISSUE THREAD

**Issue by dpaeme** [iina/iina#4790]
**System and IINA version:**

- macOS Sonoma 14.2.1 (23C71)
- IINA 1.3.4 Build 140

**Expected behavior:**
When moving to the next track in the playlist, it should start from 00:00

**Actual behavior:**
When moving to the next track in the playlist, it starts playing from a (random?) timestamp sometimes.

<details>
<summary><b>Crash report:</b></summary>

```
```
</details>

<br>

<details>
<summary><b>mpv log:</b></summary>

```
```
</details>

<br>

**Steps to reproduce:**

- [ ] MPV does not have this problem.

**How often does this happen?**

**Comment by svobs** [comment]
See if the `Resume last playback position` option is enabled in `Settings` > `General` > `Behavior`. That is the most likely possible cause of this behavior.

**Comment by dpaeme** [comment]
Yep, looks like it's the cause indeed. And while the behaviour makes sense, especially for movies/video, I'd kind of expect it to work differently in a music playlist scenario. 

It doesn't seem too logical that IINA would skip to a past timestamp when moving trough a list.

Thanks for the heads up!

**Comment by svobs** [comment]
> It doesn't seem too logical that IINA would skip to a past timestamp when moving trough a list.

That does make sense...

Since IINA is built around mpv, this checkbox actually enables [mpv's "watch later" feature](https://mpv.io/manual/stable/#resuming-playback). This feature was supposedly designed for users who wanted to save their position and settings for the file when they quit, then restore them when the file is opened again. But that asymmetry in how it works (save only by quitting the app, but restoring by opening the file) makes it confusing to use. And it has other significant shortcomings too.

Probably it would be better to disable this feature by default, and have a clearer label about how exactly it works - preferably paired with new solution like what has been requested via #4012.
